Testing branch

mikethreels opened the pull request:
## Milestone: Tic-Tac-Toe Testing

### when updating the project the following points have been added.
- [x] refactored code to make it easier to work with.
- [x] created separate modules.
- [x] separated the DOM-manipulation from the logic
- [x] created unit testing for the functions.

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi @mikethreels and @carlosprietobarron, great job with this project :+1: PR is detailed with title and a short summary :100: 

- Let us kindly separate our test for each class or module. Currently, we have one big file with all the test
![testt](https://user-images.githubusercontent.com/46388113/96289447-9dceef00-0fd4-11eb-8836-4f9d23dc4ea8.png)
- Since this is a testing project, kindly add the methods tested in the README file and a screenshot of the test

[Optional]
- There is a hack on the logic, one player can select all the box without the other player selecting. So the first player selects `pos 1`, the second player also selects the same position but he gets an alert message (which should be a modal). The play switch back to the first player

Once these are addressed, kindly submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

mikethreels commented:
## Changelog

### when addressing the requested changes the following points have been adjusted.
- [x] separated our tests for each module
- [x] add the methods tested in the README file 
- [x] added a screenshot of the test
- [x] fixed the hack the previous TSE mentioned in his review.

MarioDena reviewed (approved):
Hello well done implementing the requested changes! 🥇 

Project approved! 🟢 

Cheers!
